phobos/sys/traits.d

```diff
@@ -2580,6 +2584,492 @@ template isSameType(T)
+    static assert( isDelegate!(PropertyType!del));
+    static assert(!isDelegate!(typeof(&del)));
+    static assert( is(SymbolType!del == delegate));
+    static assert( is(PropertyType!del == delegate));
+    static assert(!is(typeof(&del) == delegate));
+
+    @property static int prop() { return 0; }
+    static assert(!isDelegate!(SymbolType!prop));
+    static assert(!isDelegate!(PropertyType!prop));
+    static assert(!isDelegate!(typeof(&prop)));
+    static assert(!is(SymbolType!prop == delegate));
+    static assert(!is(PropertyType!prop == delegate));
+    static assert(!is(typeof(&prop) == delegate));
+
+    @property int propWithContext() { return var; }
+    static assert(!isDelegate!(SymbolType!propWithContext));
+    static assert(!isDelegate!(PropertyType!propWithContext));
+    static assert( isDelegate!(typeof(&propWithContext)));
+    static assert(!is(SymbolType!propWithContext == delegate));
+    static assert(!is(PropertyType!propWithContext == delegate));
+    static assert( is(typeof(&propWithContext) == delegate));
+
+    static int function() propFuncPtr() @property { return null; }
+    static assert(!isDelegate!(SymbolType!propFuncPtr));
+    static assert(!isDelegate!(PropertyType!propFuncPtr));
+    static assert(!isDelegate!(typeof(&propFuncPtr)));
+    static assert(!is(SymbolType!propFuncPtr == delegate));
+    static assert(!is(PropertyType!propFuncPtr == delegate));
+    static assert(!is(typeof(&propFuncPtr) == delegate));
+
+    static int delegate() propDel() @property { return null; }
+    static assert(!isDelegate!(SymbolType!propDel));
+    static assert( isDelegate!(PropertyType!propDel));
+    static assert(!isDelegate!(typeof(&propDel)));
+    static assert(!is(SymbolType!propDel == delegate));
+    static assert( is(PropertyType!propDel == delegate));
+    static assert(!is(typeof(&propDel) == delegate));
+
+    static struct S
+    {
+        void foo() {}
+    }
+    static assert(!isDelegate!(SymbolType!(S.foo)));
+    static assert(!isDelegate!(typeof(&S.foo)));
+    static assert( isDelegate!(typeof(&S.init.foo)));
+    static assert(!is(SymbolType!(S.foo) == delegate));
+    static assert(!is(typeof(&S.foo) == delegate));
+    static assert( is(typeof(&S.init.foo) == delegate));
+
+    struct HasContext
+    {
+        void foo() { ++var; }
+    }
+    static assert(!isDelegate!(SymbolType!(S.foo)));
+    static assert(!isDelegate!(typeof(&S.foo)));
+    static assert( isDelegate!(typeof(&S.init.foo)));
+    static assert(!is(SymbolType!(S.foo) == delegate));
+    static assert(!is(typeof(&S.foo) == delegate));
+    static assert( is(typeof(&S.init.foo) == delegate));
+}
+
+/++
+    Evaluates to $(K_TRUE) if the given type is a function (and to $(K_FALSE)
+    otherwise).
+
+    This is equivalent to $(D is(T == function)), so most code shouldn't use
+    it. It's intended to be used in conjunction with templates that take a
+    template predicate - such as those in phobos.sys.meta.
```

Comment:
do we want to provide a link to `phobos.sys.meta` in the documentation here? Ditto for all